subiquity/models/filesystem.py

```diff
@@ -848,7 +864,9 @@ def _can_PARTITION(self):
             return False
         if self.free_for_partitions <= 0:
             return False
-        if self.ptable == 'vtoc' and len(self._partitions) >= 3:
+        # we only use msdos partition tables with FBA dasds, which
+        # only support 3 partitions.
+        if self.ptable in ['vtoc', 'msdos'] and len(self._partitions) >= 3:
```

Comment:
Hm.... if we are resuing partitions, on any drive on any architecture we may hit msdos. Which may have 4 or more partitions (thanks to the extended partitions).

I guess that would be a TODO to fix, if we ever want to support msdos more fully.

Reply:
Don't we know DASD type? if not, you could scope the 3 partition limit to arch=s390x to avoid tripping existing MSDOS layouts on non-s390x.

Reply:
there is `self.dasd`.... but i can't tell if it is set when this code runs. @mwhudson ? is self.dasd live at this point, and can we do `if self.dasd` check instead of `self.ptable in ['vtoc', 'msdos']` given that only these two are allowed on dasd. gpt just blows up on them.